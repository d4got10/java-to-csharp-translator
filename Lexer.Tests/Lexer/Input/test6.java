public class Transpose {

    public static void main(String[] args) {
        int row = 2, column = 3;
        int[][] matrix = { {2, 3, 4}, {5, 6, 4} };

        display(matrix);

        int[][] transpose = new int[column][row];
        for(int i = 0; i < row; i++) {
            for (int j = 0; j < column; j++) {
                transpose[j][i] = matrix[i][j];
            }
        }

        display(transpose);
    }

    public static void display(int[][] matrix) {
        System.out.println("The matrix is: ");
        for(int[] row : matrix) {
            for (int column : row) {
                System.out.print(column + "    ");
            }
            System.out.println();
        }
    }
}
