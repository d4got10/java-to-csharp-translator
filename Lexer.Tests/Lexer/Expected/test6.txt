AccessModifier public 1:1
Keyword class 1:8
Identifier Transpose 1:14
OpenBracket { 1:24
AccessModifier public 3:5
Keyword static 3:12
Type void 3:19
Identifier main 3:24
Operator ( 3:28
Identifier String 3:29
Operator [ 3:35
Operator ] 3:36
Identifier args 3:38
Operator ) 3:42
OpenBracket { 3:44
Type int 4:9
Identifier row 4:13
Operator = 4:17
Value 2 4:19
Comma , 4:20
Identifier column 4:22
Operator = 4:29
Value 3 4:31
Semicolon ; 4:32
Type int 5:9
Operator [ 5:12
Operator ] 5:13
Operator [ 5:14
Operator ] 5:15
Identifier matrix 5:17
Operator = 5:24
OpenBracket { 5:26
OpenBracket { 5:28
Value 2 5:29
Comma , 5:30
Value 3 5:32
Comma , 5:33
Value 4 5:35
CloseBracket } 5:36
Comma , 5:37
OpenBracket { 5:39
Value 5 5:40
Comma , 5:41
Value 6 5:43
Comma , 5:44
Value 4 5:46
CloseBracket } 5:47
CloseBracket } 5:49
Semicolon ; 5:50
Identifier display 7:9
Operator ( 7:16
Identifier matrix 7:17
Operator ) 7:23
Semicolon ; 7:24
Type int 9:9
Operator [ 9:12
Operator ] 9:13
Operator [ 9:14
Operator ] 9:15
Identifier transpose 9:17
Operator = 9:27
Keyword new 9:29
Type int 9:33
Operator [ 9:36
Identifier column 9:37
Operator ] 9:43
Operator [ 9:44
Identifier row 9:45
Operator ] 9:48
Semicolon ; 9:49
Keyword for 10:9
Operator ( 10:12
Type int 10:13
Identifier i 10:17
Operator = 10:19
Value 0 10:21
Semicolon ; 10:22
Identifier i 10:24
Operator < 10:26
Identifier row 10:28
Semicolon ; 10:31
Identifier i 10:33
Operator ++ 10:34
Operator ) 10:36
OpenBracket { 10:38
Keyword for 11:13
Operator ( 11:17
Type int 11:18
Identifier j 11:22
Operator = 11:24
Value 0 11:26
Semicolon ; 11:27
Identifier j 11:29
Operator < 11:31
Identifier column 11:33
Semicolon ; 11:39
Identifier j 11:41
Operator ++ 11:42
Operator ) 11:44
OpenBracket { 11:46
Identifier transpose 12:17
Operator [ 12:26
Identifier j 12:27
Operator ] 12:28
Operator [ 12:29
Identifier i 12:30
Operator ] 12:31
Operator = 12:33
Identifier matrix 12:35
Operator [ 12:41
Identifier i 12:42
Operator ] 12:43
Operator [ 12:44
Identifier j 12:45
Operator ] 12:46
Semicolon ; 12:47
CloseBracket } 13:13
CloseBracket } 14:9
Identifier display 16:9
Operator ( 16:16
Identifier transpose 16:17
Operator ) 16:26
Semicolon ; 16:27
CloseBracket } 17:5
AccessModifier public 19:5
Keyword static 19:12
Type void 19:19
Identifier display 19:24
Operator ( 19:31
Type int 19:32
Operator [ 19:35
Operator ] 19:36
Operator [ 19:37
Operator ] 19:38
Identifier matrix 19:40
Operator ) 19:46
OpenBracket { 19:48
Identifier System 20:9
Operator . 20:10
Identifier out 20:13
Operator . 20:14
Identifier println 20:21
Operator ( 20:27
Value "The matrix is: " 20:28
Operator ) 20:29
Semicolon ; 20:30
Keyword for 21:9
Operator ( 21:12
Type int 21:13
Operator [ 21:16
Operator ] 21:17
Identifier row 21:19
Identifier : 21:23
Identifier matrix 21:25
Operator ) 21:31
OpenBracket { 21:33
Keyword for 22:13
Operator ( 22:17
Type int 22:18
Identifier column 22:22
Identifier : 22:29
Identifier row 22:31
Operator ) 22:34
OpenBracket { 22:36
Identifier System 23:17
Operator . 23:18
Identifier out 23:21
Operator . 23:22
Identifier print 23:27
Operator ( 23:33
Identifier column 23:34
Operator + 23:41
Value "    " 23:43
Operator ) 23:44
Semicolon ; 23:45
CloseBracket } 24:13
Identifier System 25:13
Operator . 25:14
Identifier out 25:17
Operator . 25:18
Identifier println 25:25
Operator ( 25:31
Operator ) 25:32
Semicolon ; 25:33
CloseBracket } 26:9
CloseBracket } 27:5
CloseBracket } 28:1
